docker/Dockerfile_ci

```diff
@@ -1,84 +1,113 @@
+      && rm -f yaml-0.1.7.tar.gz \
+      && cd /usr/local/yaml-0.1.7 \
+      && ./configure \
+      && make \
+      && make install \
+      # Install git
+      && apt-get install -y \
+            git \
+      && git config --global user.email "bob@builder.com" \
+      &&  git config --global user.name "Bob the Builder" \
+      # Install gradle
+      && wget -q https://services.gradle.org/distributions/gradle-$GRADLE_VERSION-bin.zip \
+      && unzip -q gradle-$GRADLE_VERSION-bin.zip -d /usr/local \
+      && rm -f gradle-$GRADLE_VERSION-bin.zip \
+      && ln -s /usr/local/gradle-$GRADLE_VERSION/bin/gradle /usr/bin/gradle
+
+
 
 # Setup reviewsite environment
-RUN apk update \
- \
- # Install PHP
- && apk add \
-      php7 \
-      php7-ctype \
-      php7-dom \
-      php7-json \
-      php7-mbstring \
-      php7-openssl \
-      php7-pdo \
-      php7-pdo_sqlite \
-      php7-phar \
-      php7-session \
-      php7-sqlite3 \
-      php7-tokenizer \
-      php7-xml \
-      php7-xmlwriter \
-      php7-zlib \
-      php7-curl \
- \
- # Install Composer
- && wget -q https://getcomposer.org/installer -O composer-setup.php \
- && php composer-setup.php --install-dir=/usr/local/bin --filename=composer \
- && rm composer-setup.php
+RUN LC_ALL=C.UTF-8 add-apt-repository -y ppa:ondrej/php \
+      && apt-get update \
+      && apt-get install -y \
+            php7.0 \
+            php7.0-xml \
+            php7.0-mbstring \
+            php7.0-sqlite \
+            php7.0-zip \
+            php7.0-curl \
+      # Install composer
+      && wget -q https://getcomposer.org/installer -O composer-setup.php \
+      && php composer-setup.php --install-dir=/usr/local/bin --filename=composer \
+      && rm composer-setup.php 
 
 # Setup test environment
 RUN \
- # Install PHPUnit
- wget -q https://phar.phpunit.de/phpunit.phar -O /usr/local/bin/phpunit \
- && chmod +x /usr/local/bin/phpunit \
- \
- # Install nosetests
- && pip3 install nose
-
-RUN apk add \
-      # Install chromium and chromedriver
-      chromium \
-      chromium-chromedriver \
+      # Install PHPUnit
+      wget -q https://phar.phpunit.de/phpunit-6.0.0.phar -O /usr/local/bin/phpunit \
+      && chmod +x /usr/local/bin/phpunit \
+      # Install nosetests
+      && pip3 install nose \
+      # Install chromeium/chromedriver
+      && apt-get install -y \
+            chromium-browser \
+            chromium-chromedriver \
       #Install selenium
-      && wget -q https://selenium-release.storage.googleapis.com/3.12/selenium-server-standalone-3.12.0.jar \
-      && mv selenium-server-standalone-3.12.0.jar /usr/local/bin/selenium-server-standalone-3.12.0.jar
+            && wget -q https://selenium-release.storage.googleapis.com/3.12/selenium-server-standalone-3.12.0.jar \
+          && mv selenium-server-standalone-3.12.0.jar /usr/local/bin/selenium-server-standalone-3.12.0.jar \
+          && ln /usr/lib/chromium-browser/chromedriver /usr/local/bin/chromedriver
+
+# Cleanup
```

Comment:
This does not make sense. It does not make the image smaller, but larger! Every Docker command creates a new layer with the diff to the underlying layers. If you want to remove things, execute a cleanup at the end of a command that installs stuff. Then the cleanup happens before the diff is generated and the diff might actually get smaller.